ambry-store/src/test/java/com.github.ambry.store/IndexTest.java

```diff
@@ -782,6 +782,36 @@ public void findEntriesSinceOnRestartTest() throws IOException, StoreException {
   }
 
   /**
+   * Tests {@link PersistentIndex#getLogSegmentsNotInJournal()}
+   * @throws InterruptedException
+   * @throws StoreException
+   * @throws IOException
+   */
+  @Test
+  public void getLogSegmentsNotInJournalTest() throws InterruptedException, IOException, StoreException {
+    if (isLogSegmented) {
+      testGetLogSegmentsNotInJournal(2, 2 * MAX_IN_MEM_ELEMENTS, false);
```

Comment:
I tried cleaning up the index and retrying with much more values for "maxLogSegmentsToBeIgnored", but since CuratedLogIndexState hasn't been designed to clean up logs and do retries within a single test, don't want to get too detailed over this. So, felt this would suffice.